kasa/protocol.py

```diff
@@ -41,6 +90,7 @@ def __init__(
         *,
         port: Optional[int] = None,
         credentials: Optional[Credentials] = None,
+        transport: Optional[BaseTransport] = None,
```

Comment:
I realized that not all implementations accept `transport` (and at least `TPLinkSmartHomeProtocol` has `timeout` added).

Maybe we should enforce consistent parameters all across the board (by adding a test for the ctors?)?

Reply:
I agree it makes sense.  Would prefer to add it in a subsequent PR as it'll be easier to see what's going and we may want to convert the legacy TPLinkSmartHomeProtocol class to use the IotProtocol with it's own XoR transport broken out.  Am thinking about the request for re-use etc.

Reply:
Agreed, let's get this one merged and follow-up with smaller and easier to review PRs :-)